rust/src/service.rs

```diff
@@ -60,10 +60,10 @@ impl Service {
     //
     // when we fail to handle a message, we should send back and error
     // so that the participant is informed. One problem currently is
-    // that the `validate` methods don't return specific errors.
+    // that the `handle` methods don't return specific errors.
```

Comment:
on an important side note: the coordinator **must never** inform a participant about the specific reason why message validation failed. that would open gates to various attacks. instead the coordinator must ignore that message, but it can/should log the failure internally.
it is ok to, for example, send an ACK to the participant if the message was received. it might also send a generic failure message if validation fails, but there is no reason to do so and it's better to not open up to potential attacks.